File: Python/programmers/etc/level1/크기가_작은_부분_문자열.py
def solution(t, p):
    answer = 0
    T, P = len(t), len(p)
    for i in range(T-P+1):
        tmp = t[i:i+P]
        if tmp <= p:
            answer += 1
    return answer


t, p = "3141592", "271"
print(solution(t, p)) # 2

t, p = "500220839878", "7"
print(solution(t, p)) # 8

t, p = "10203", "15"
print(solution(t, p)) # 3

File: Java/programmers/etc/level1/크기가작은부분문자열.java
package CodeTest.Java.programmers.etc.level1;

public class 크기가작은부분문자열 {
    public int solution(String t, String p) {
        int answer = 0;
        int T = t.length(), P = p.length();
        long n = Long.parseLong(p);
        for (int i=0; i<=T-P; i++) {
            long tmp = Long.parseLong(t.substring(i, i+P));
            if (tmp <= n) answer++;
        }
        return answer;
    }
}
